### PR TITLE
feat/create_order_utils

### PR DESCRIPTION
**Pull request by sparqet**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

- Implemented create_order in 

## Does this introduce a breaking change?
No

### CODE REVIEW

File: src/order/order_utils.cairo [hunk 1]
```diff
@@ -41,99 +41,100 @@ fn create_order( //TODO and fix when wnt is implememted
...
-
-// event_emitter.emit_order_created(key, order);
-
-// key
+    account_utils::validate_account(account);
+    referral_utils::set_trader_referral_code(referral_storage, account, params.referral_code);
+
+    let mut initial_collateral_delta_amount = 0;
+
+    let wnt = token_utils::fee_token(data_store);
```
Comment (zarboq) [inline src/order/order_utils.cairo:51]:
```suggestion
    let fee_token = token_utils::fee_token(data_store);
```

File: src/order/order_utils.cairo [hunk 2]
```diff
@@ -41,99 +41,100 @@ fn create_order( //TODO and fix when wnt is implememted
...
+        || params.order_type == OrderType::LimitDecrease
+        || params.order_type == OrderType::StopLossDecrease) {
+        // for decrease orders, the initialCollateralDeltaAmount is based on the passed in value
+        initial_collateral_delta_amount = params.initial_collateral_delta_amount;
+    } else {
+        OrderError::ORDER_TYPE_CANNOT_BE_CREATED(params.order_type);
+    }
+
+    if (should_record_separate_execution_fee_transfer) {
+        let wnt_amount = order_vault.record_transfer_in(wnt);
```
Comment (zarboq) [inline src/order/order_utils.cairo:82]:
```suggestion
        let fee_token_amount = order_vault.record_transfer_in(fee_token);
```

**Review by zarboq — CHANGES_REQUESTED** [review]

**Review by zarboq — APPROVED** [review]

**Comment by Sk8erboi84** [comment]
Good to merge after tests

**Review by Sk8erboi84 — APPROVED** [review]
Good to merge after tests

**Pull request merged**